src/axom/inlet/Table.hpp

```diff
@@ -921,9 +995,119 @@ class Table : public Verifiable
+   *****************************************************************************
+   */
+  template <typename T>
+  typename std::enable_if<detail::is_inlet_primitive<T>::value, bool>::type
+  getArray(std::unordered_map<int, T>& map) const
+  {
+    map.clear();
+    if(!axom::utilities::string::endsWith(m_name, CONTAINER_GROUP_NAME))
+    {
+      return false;
+    }
+    for(auto& item : m_fieldChildren)
+    {
+      auto pos = item.first.find_last_of("/");
+      int index = std::stoi(item.first.substr(pos + 1));
+      map[index] = item.second->get<T>();
+    }
+    return true;
+  }
+
+  /*!
+   *****************************************************************************
+   * \brief Get an array represented as an unordered map from the input file
+   * of user-defined type
+   *
+   * \param [out] map Unordered map to be populated with array contents
+   *
+   * \return Whether or not the array was found
+   *****************************************************************************
+   */
+  template <typename T>
+  typename std::enable_if<!detail::is_inlet_primitive<T>::value, bool>::type
+  getArray(std::unordered_map<int, T>& map) const
+  {
+    for(const auto& indexLabel : containerIndices())
+    {
+      map[std::stoi(indexLabel)] = getTable(indexLabel).get<T>();
+    }
+    return true;
+  }
+
+  /*!
+   *****************************************************************************
+   * \brief Get a dict represented as an unordered map from the input file
+   * of primitive type
+   *
+   * \param [out] map Unordered map to be populated with dictionary contents
+   *
+   * \return Whether or not the dictionary was found
+   *****************************************************************************
+   */
+  template <typename T>
+  typename std::enable_if<detail::is_inlet_primitive<T>::value, bool>::type
+  getDict(std::unordered_map<std::string, T>& map) const
+  {
+    map.clear();
+    if(!axom::utilities::string::endsWith(m_name, CONTAINER_GROUP_NAME))
+    {
+      return false;
+    }
+    for(auto& item : m_fieldChildren)
+    {
+      auto pos = item.first.find_last_of("/");
+      auto index = item.first.substr(pos + 1);
+      map[index] = item.second->get<T>();
+    }
+    return true;
+  }
+
+  /*!
+   *****************************************************************************
+   * \brief Get a dict represented as an unordered map from the input file
+   * of user-defined type
+   *
+   * \param [out] map Unordered map to be populated with dictionary contents
+   *
+   * \return Whether or not the dictionary was found
+   *****************************************************************************
+   */
+  template <typename T>
+  typename std::enable_if<!detail::is_inlet_primitive<T>::value, bool>::type
+  getDict(std::unordered_map<std::string, T>& map) const
+  {
+    for(const auto& indexLabel : containerIndices())
+    {
+      map[indexLabel] = getTable(indexLabel).get<T>();
+    }
+    return true;
```

Comment:
This always returns true. Is there something in the templating that i am missing that returns false?

Reply:
This is the version for dicts of user-defined type; the version for dicts of primitives can return false (in which case it will just `SLIC_ERROR`.  This function can still fail (if `containerIndices` fails with a `SLIC_ERROR`), but I will add a check to this function for consistency.